Execution failed for task ':checkDebugDuplicateClasses'.

C:\Users\HOMEPC\Desktop\godot-3.1>scons p=android target=release tools=no disable_3d=yes target=release tools=no module_bmp_enabled=no module_bullet_enabled=no module_csg_enabled=no module_dds_enabled=no module_enet_enabled=no module_etc_enabled=no module_gdnative_enabled=no module_gridmap_enabled=no module_hdr_enabled=no module_mbedtls_enabled=no module_mobile_vr_enabled=no module_opus_enabled=no module_pvr_enabled=no module_recast_enabled=no module_regex_enabled=no module_squish_enabled=no module_tga_enabled=no module_thekla_unwrap_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_vorbis_enabled=no module_webm_enabled=no module_websocket_enabled=no optimize=size
scons: Reading SConscript files ...

scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly
File "C:\Users\HOMEPC\Desktop\godot-3.1\SConstruct", line 3, in <module>
Building for Android (armv7) (with NEON)
Using NDK unified headers

scons: warning: Calling missing SConscript without error is deprecated.
Transition by adding must_exist=0 to SConscript calls.
Missing SConscript 'modules\GodotFAN\SCsub'
File "C:\Users\HOMEPC\Desktop\godot-3.1\modules\SCsub", line 17, in <module>
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
[ 95%] [91mLinking Shared Library [95m==> [93mbin\libgodot.android.opt.armv7.neon.so[0m
[100%] progress_finish(["progress_finish"], [])
[100%] scons: done building targets.

C:\Users\HOMEPC\Desktop\godot-3.1>.\gradlew build
'.\gradlew' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\HOMEPC\Desktop\godot-3.1>cd platform/android/java

C:\Users\HOMEPC\Desktop\godot-3.1\platform\android\java>.\gradlew build

> Configure project :
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html

> Task :checkDebugDuplicateClasses FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkDebugDuplicateClasses'.
> 1 exception was raised by workers:
  java.lang.RuntimeException: java.lang.RuntimeException: Duplicate class android.support.v4.app.LoaderManager found in modules classes.jar (com.android.support:loader:28.0.0) and classes.jar (com.android.support:support-fragment:25.2.0)
  Duplicate class android.support.v4.app.LoaderManager$LoaderCallbacks found in modules classes.jar (com.android.support:loader:28.0.0) and classes.jar (com.android.support:support-fragment:25.2.0)
  Duplicate class android.support.v4.app.LoaderManagerImpl found in modules classes.jar (com.android.support:loader:28.0.0) and classes.jar (com.android.support:support-fragment:25.2.0)
  Duplicate class android.support.v4.app.LoaderManagerImpl$LoaderInfo found in modules classes.jar (com.android.support:loader:28.0.0) and classes.jar (com.android.support:support-fragment:25.2.0)
  Duplicate class android.support.v4.view.NestedScrollingChildHelper found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.view.NestedScrollingParentHelper found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.AutoScrollHelper found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.AutoScrollHelper$ClampedScroller found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.AutoScrollHelper$ScrollAnimationRunnable found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.ContentLoadingProgressBar found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.ContentLoadingProgressBar$1 found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.ContentLoadingProgressBar$2 found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.ListViewAutoScrollHelper found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.NestedScrollView found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.NestedScrollView$AccessibilityDelegate found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.NestedScrollView$OnScrollChangeListener found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.NestedScrollView$SavedState found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)
  Duplicate class android.support.v4.widget.NestedScrollView$SavedState$1 found in modules classes.jar (com.android.support:support-compat:28.0.0) and classes.jar (com.android.support:support-core-ui:27.1.1)

  Go to the documentation to learn how to <a href="d.android.com/r/tools/classpath-sync-errors">Fix dependency resolution errors</a>.


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
17 actionable tasks: 1 executed, 16 up-to-date
<-------------> 0% WAITING
> IDLE
> IDLE

--- a/README.md
+++ b/README.md
@@ -1,11 +1,10 @@
-# Facebook Audience Network *GODOT MODULE*
+# Facebook Audience Network *GODOT Plugin compatible with godot >= 3.2.1 new plugin system*
 
-## Using the module
+## Using the plugin
 
-`clone` the repo to your godot source code modules file 
-`godot_source/modules`
+`clone` the repo to your `your_godot_project/android` folder or download and rename The folder to GodotFAN , after installing [Android Build Template](https://docs.godotengine.org/en/stable/getting_started/workflow/export/android_custom_build.html)
+*the new Godot plugin system doesn't require compiling the source code*
 
-*use `scons platform=android` to build android export templates*
 
 ### Initialize the module
 in your `project.godot` add
@@ -13,14 +12,14 @@ in your `project.godot` add
 [android]
 modules="org/godotengine/godot/GodotFAN"
 ```
+Or from your project settings (Android modules section)
+
 ## API 
 ```
 var facebookAds
 func _ready()-> void:
-	
 	facebookAds = Engine.get_singleton("GodotFAN")
 	facebookAds.FacebookAdsInit("YOUR_INTERSTITIAL_PLACEMENT_id","YOUR_REWARDED_VIDEO_PLACEMENT_id")
-
 ```
 
 ### Calling the interstitial and rewarded video ads

--- a/config.py
+++ b/config.py
@@ -1,8 +1,0 @@
-def can_build(env,plat):
-    return plat=="android"
-
-def configure(env):
-    if env['platform'] == 'android':
-        env.android_add_java_dir("src")
-        env.android_add_dependency("compile 'com.android.support:recyclerview-v7:27.1.1'")
-        env.android_add_dependency("compile 'com.facebook.android:audience-network-sdk:5.+'")

--- a/gradle.conf
+++ b/gradle.conf
@@ -1,0 +1,3 @@
+[dependencies]
+    implementation 'com.android.support:recyclerview-v7:27.1.1'
+    implementation 'com.facebook.android:audience-network-sdk:5.+'

